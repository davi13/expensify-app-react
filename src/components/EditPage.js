import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';


export class EditPage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense(expense);
        this.props.history.push('/');
    }
    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit} />
                <button onClick={() => {
                    this.props.dispatch(removeExpense({ id: props.expense.id }));
                    this.props.history.push('/')

                }}> Remove </button>
            </div>
        );
    }
}
const mapDispatchToProps = (dispatch) => ({
    editExpense: dispatch(editExpense(expense))
})

export default connect(mapDispatchToProps)(EditPage);