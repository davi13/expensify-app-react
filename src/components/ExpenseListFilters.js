import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';


export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
        console.log(this.state.calendarFocused);
    }
    render() {
        return (
            <div>
                <input
                    type="text"
                    value={this.props.filters.text}
                    onChange={(e) => {
                        this.props.dispatch(setTextFilter(e.target.value));
                    }}
                />
                <select
                    value={this.props.filters.sortBy}
                    onChange={(e) => {
                        e.target.value === 'date' ? this.props.dispatch(sortByDate(e.target.value)) : this.props.dispatch(sortByAmount(e.target.value));
                    }}
                >
                    <option value="date">Date</option>
                    <option value="amount">amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
        );
    }
};
const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };

};

export default connect(mapStateToProps)(ExpenseListFilters);