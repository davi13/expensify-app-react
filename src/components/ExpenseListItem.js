import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>
            {numeral(amount / 100).format('$0,0.00')}
            -
        {moment(createdAt).format('Do MMMM YYYY')}</p>
        {/*<button onClick={() => {
            dispatch(removeExpense({ id }));

        }}> Remove </button>*/}
    </div>
);
export default ExpenseListItem;