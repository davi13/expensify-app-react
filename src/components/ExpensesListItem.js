import React from 'react';
import { connect } from 'react-redux';
const ExpenseListItem = (props) => (
    <div>
        {props.expenses.map((expense, key) => (
            <li key={key}>
                {expense.amount}
                {expense.description}
                {expense.createAt}
            </li>
        )
        )}
    </div>


);
const mapStateToProps = (state) => {
    return {
        expenses: state.expenses
    }
}
export default connect(mapStateToProps)(ExpenseListItem)