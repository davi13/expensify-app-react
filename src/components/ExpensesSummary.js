import React from 'react'
import numeral from 'numeral'
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    //const totalAmount = props.expenses.map((expense) => expense.amount).reduce((sum, value) => sum + value, 0);

    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')
    return (
        <div>
            <span>
                {`Viewing  ${expenseCount} ${expenseWord} totalling ${formattedExpensesTotal}`}

            </span>
        </div>
    )
}
const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}
export default connect(mapStateToProps)(ExpensesSummary);