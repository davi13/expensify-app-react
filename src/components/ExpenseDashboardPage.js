import React from 'react';
import ExpensesSmmary from '../components/ExpensesSummary'
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';


const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
        <ExpensesSmmary />
    </div>
);

export default ExpenseDashboardPage;
