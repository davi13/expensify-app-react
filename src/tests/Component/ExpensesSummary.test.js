import React from 'react'
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses';

test('should rendly correctly expensesSummary', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});
test('should rendly correctly expensesSummary with mutiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

