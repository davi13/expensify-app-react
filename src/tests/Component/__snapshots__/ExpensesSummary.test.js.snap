// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendly correctly expensesSummary 1`] = `
<div>
  <span>
    Viewing  1 expense totalling $2.35
  </span>
</div>
`;

exports[`should rendly correctly expensesSummary with mutiple expenses 1`] = `
<div>
  <span>
    Viewing  2 expenses totalling $2.35
  </span>
</div>
`;
